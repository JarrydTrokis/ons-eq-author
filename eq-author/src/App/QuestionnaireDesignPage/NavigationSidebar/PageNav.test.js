import React from "react";
import { shallow } from "enzyme";
import PageNav from "./PageNav";

describe("PageNav", () => {
  let component, handleDelete;

  const questionnaire = { id: "1", title: "Questionnaire" };
  const pages = [
    { id: "2", title: "Page" },
    { id: "4", title: "Confirmation Page", __typename: "QuestionConfirmation" },
  ];
  const section = { id: "3", title: "Section", pages: pages, number: 1 };

  beforeEach(() => {
    handleDelete = jest.fn(() => Promise.resolve());

    component = shallow(
      <PageNav
        questionnaire={questionnaire}
        section={section}
        onDelete={handleDelete}
      />
    );
  });

  it("should render", () => {
    expect(component).toMatchSnapshot();
  });
});
