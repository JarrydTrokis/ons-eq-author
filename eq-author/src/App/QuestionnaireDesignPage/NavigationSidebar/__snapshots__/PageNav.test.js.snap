// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNav should render 1`] = `
<TransitionGroup
  childFactory={[Function]}
  component={[Function]}
>
  <NavItemTransition
    in={false}
    key="2"
    onEntered={[Function]}
    timeout={300}
  >
    <withRouter(WrappedComponent)
      page={
        Object {
          "id": "2",
          "title": "Page",
        }
      }
      questionnaireId="1"
      sectionId="3"
    />
  </NavItemTransition>
  <NavItemTransition
    in={false}
    key="confirmation-4"
    onEntered={[Function]}
    timeout={300}
  >
    <withRouter(UnwrappedPageConfirmationNavItem)
      confirmation={
        Object {
          "__typename": "QuestionConfirmation",
          "id": "4",
          "title": "Confirmation Page",
        }
      }
      page={
        Object {
          "id": "2",
          "title": "Page",
        }
      }
      questionnaireId="1"
      sectionId="3"
    />
  </NavItemTransition>
</TransitionGroup>
`;
